src/coreclr/vm/codeman.cpp

```diff
@@ -4475,7 +4493,6 @@ BOOL ExecutionManager::IsManagedCodeWithLock(PCODE currentPC)
         GC_NOTRIGGER;
     } CONTRACTL_END;
 
-    ReaderLockHolder rlh;
```

Comment:
Why is it ok to remove this lock?

Reply:
I see that `GetRangeSection` will unconditionally take the lock, but it will return a RangeSection, that itself isn't protected from unload. Please restore the original structure of how the locks are created.

Reply:
As I can understand your comment, the function of rlh there is not for protecting the RS list from possibly inconsistence but instead for protecting an RS using in IsManagedCodeWorker from deletion.

In my proposal, as described in my answer to you at https://github.com/dotnet/runtime/pull/68607#discussion_r866693016 rlh and wlh are used only for maintain RS storage consistency, but the schema leaves us an opportunity to restore the RS instance protecting lock structure. In short, I can add **wlh** constructor with an no-op argument just for stopping all the writers at the spot until wlh release. So in current lines we will wait till current (if any) writer finishes, take writer's lock, and go to IsManagedCodeWorker which can freely use GetRangeSection as soon as readers can go independently of writers.

Do you see that in perspective of https://github.com/dotnet/runtime/pull/68607#discussion_r866693016 this approach of maintaining RS instance protection is correct?